File: src/interfaces/objectInterfaces/IMite.ts
/** @format */

import { IMiteType } from '../templateLiteralUnionTypes/IMiteType'

//  IMite.ts
//  MEGADocker
//  an interface that represents a Mite - a building block generic
//  Created by George Georgulas IV on 1/26/19.
//  Copyright © 2019-2021 The MegaDocker Group. All rights reserved.

/**
 * interface for IMite
 * @type which type of IMite this IMite is
 * @miteIndex uniquie identifier for this IMite
 * @miteString the IMite's data
 */
export interface IMite {
  type: IMiteType
  miteIndex: number
  miteString: string
}

File: src/interfaces/objectInterfaces/IMite.rs
//  IMite.rs
//  MEGADocker
//  a struct that represents a Mite - a building block generic
//  Created by George Georgulas IV on 1/26/19.
//  Copyright © 2019-2022 The MegaDocker Group. All rights reserved.

use src::structs::object_interfaces::IMiteType;

pub struct Mite {
  pub type_: IMiteType,
  pub mite_index: u32,
  pub mite_string: String,
}
